test/Field/FieldAxioms.hs

```diff
@@ -0,0 +1,99 @@
+prop_addAssociative _ x y z = (x `add` y) `add` z === x `add` (y `add` z)
+
+prop_addCommutative :: (Field f, Show f, Eq f) => Proxy f -> f -> f -> Property
+prop_addCommutative _ x y = x `add` y === y `add` x
+
+prop_zerAdditiveIdentity :: (Field f, Show f, Eq f) => Proxy f -> f -> Property
+prop_zerAdditiveIdentity _ x = x `add` zer === x
+-- We can get away with only doing this on one side because of
+-- prop_addCommutative, and similarly for other tests.
+
+prop_AdditiveInverse :: (Field f, Show f, Eq f) => Proxy f -> f -> Property
+prop_AdditiveInverse _ x = x `add` (neg x) === zer
+
+
+-- (f, mul, one) is a commutative monoid
+prop_mulAssociative :: (Field f, Show f, Eq f) => Proxy f -> f -> f -> f -> Property
+prop_mulAssociative _ x y z = (x `mul` y) `mul` z === x `mul` (y `mul` z)
+
+prop_mulCommutative :: (Field f, Show f, Eq f) => Proxy f -> f -> f -> Property
+prop_mulCommutative _ x y = x `mul` y === y `mul` x
+
+prop_oneMultiplicativeIdentity :: (Field f, Show f, Eq f) => Proxy f -> f -> Property
+prop_oneMultiplicativeIdentity _ x = x `mul` one === x -- && (one `mul` x === x)
+
+-- nonzero elements have inverses
+prop_nonZeroInverse :: (Field f, Show f, Eq f) => Proxy f -> f -> Property
+prop_nonZeroInverse _ x = x /= zer ==> x `mul` (inv x) === one
+
+-- mul distributes over add
+prop_mulDistributive :: (Field f, Show f, Eq f) => Proxy f -> f -> f -> f -> Property
+prop_mulDistributive _ x y z = x `mul` (y `add` z) === (x `mul` y) `add` (x `mul` z)
+
+-- Strictly we should check that zer /= one as well.
+
+-- If we knew the characteristic we might want to check that it was
+-- either zero or prime, but we'd need a primality test for that.
+
+-- Other properties like  0.x == 0  and  x /= 0 => (xa == xb => a == b)  follow from the axioms.
+
+
+{- QuickCheck provides things like quickCheckAll which you can use to
+   test all of the prop_... properties in a file.  It uses Template
+   Haskell, and I don't think you can get it to apply each prop_ to a
+   proxy before testing it. The runAll function below checks all of
+   the properties one by one, which is maybe not ideal.
+
+   I also tried using TypeApplications instead of proxies.  That made
+   the individual properties nicer but I still couldn't get it to
+   work with QuickCheck's TH stuff.
+-}
+
+runAllWithMaxSuccess :: (Field f, Show f, Eq f, Arbitrary f) => Int -> Proxy f -> IO ()
+runAllWithMaxSuccess n proxy = do
+  quickCheck (withMaxSuccess n (prop_addAssociative proxy))
+  quickCheck (withMaxSuccess n (prop_addCommutative proxy))
+  quickCheck (withMaxSuccess n (prop_zerAdditiveIdentity proxy))
+  quickCheck (withMaxSuccess n (prop_AdditiveInverse proxy))
+  quickCheck (withMaxSuccess n (prop_mulAssociative proxy))
+  quickCheck (withMaxSuccess n (prop_mulCommutative proxy))
+  quickCheck (withMaxSuccess n (prop_oneMultiplicativeIdentity proxy))
+  quickCheck (withMaxSuccess n (prop_nonZeroInverse proxy))
+  quickCheck (withMaxSuccess n (prop_mulDistributive proxy))
```

Comment:
There is [`conjoin`](http://hackage.haskell.org/package/QuickCheck-2.6/docs/Test-QuickCheck.html#v%3aconjoin) that allows to take the conjuction of a list of properties.